src/Dfc.CourseDirectory.Core/DataStore/CosmosDb/Queries/UpdateProviderOnboarded.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using OneOf;
+using OneOf.Types;
+
+namespace Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries
+{
+    public class UpdateProviderOnboarded : ICosmosDbQuery<OneOf<NotFound, Success>>
+    {
+        public Guid ProviderId { get; set; }
+
+        public string UpdatedBy { get; set; }
+    }
```

Comment:
Add `DateTime UpdatedDate` property on here (otherwise it's harder to test with our `IClock` stuff)

Reply:
What about injecting `IClock`? (Isn't there a .net abstraction of this now, `ISystemClock`?)

Reply:
As above, current handlers that do this have a property for the `{Created|Updated}By` and that's populated by the caller (by referencing an `IClock`) e.g. https://github.com/SkillsFundingAgency/dfc-coursedirectory/blob/main/src/Dfc.CourseDirectory.WebV2/Features/NewApprenticeshipProvider/ApprenticeshipSummary.cs#L163

Reply:
There's more than one `ISystemClock` and none of them are in the BCL

Reply:
I meant injecting `IClock` directly into the handler? What's the advantages over passing it as a param?

Reply:
Not so much that it's any better or worse, more that it's consistent with everywhere else.

Reply:
Updated